scripts/collector/sqlserver/dmaSQLServerHWSpecs.ps1

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+<#
+.SYNOPSIS
+    .
+.DESCRIPTION
+    Collects HW Specs using Get-WmiObject to be uploaded to Google Database Migration Assistant for MS SQL Server.
+.PARAMETER computerName
+    The target computer name to collect the HW specs from (Optional).
+.PARAMETER outputPath
+    The output full path of the csv that this scripts creates (Required).
+.PARAMETER pkey
+    Final Output Directory (Required).
+.PARAMETER dmaSourceId
+    DMA derived unique id (Required).
+.PARAMETER dmaManualId
+    Customer Manual Unique ID or dma manual unique id (Optional).
+.EXAMPLE
+    C:\dmaSQLServerHWSpecs.ps1 -computerName localhost -outputPath a.out -pkey pkey1 -dmaSourceId src1
+.NOTES
+    https://googlecloudplatform.github.io/database-assessment/
+#>
+param (
+	[Parameter(
+		Mandatory=$False,
+		HelpMessage="The computer name"
+	)][string]$computerName = $env:COMPUTERNAME,
+	[Parameter(
+		Mandatory=$True,
+		HelpMessage="The Output path"
+	)][string]$outputPath,
+	[Parameter(
+		Mandatory=$True,
+		HelpMessage="The pkey value"
+	)][string]$pkey,
+	[Parameter(
+		Mandatory=$True,
+		HelpMessage="The dma_source_id"
+	)][string]$dmaSourceId,
+	[Parameter(
+		Mandatory=$False,
+		HelpMessage="The dma_manual_id"
+	)][string]$dmaManualId="NA"
+)
+
+try {
+	Write-Host "Params: computer:$computerName, dma_src: $dmaSourceId, output:$outputPath"
+
```

Comment:
All logging should be done using the import:
Import-Module $PSScriptRoot\dmaCollectorCommonFunctions.psm1

You can then write messages to a log file with the standard function like this:

WriteLog -logLocation $foldername\$sqlErrorLogFile -logMessage "Checking for error messages within collection $inputFile ..." -logOperation "FILE"

Reply:
Fixed